back/src/main/java/com/baba/back/content/service/ContentService.java

```diff
@@ -206,57 +206,40 @@ private void validateRelationGroup(RelationGroup ownerRelationGroup, RelationGro
         }
     }
 
-    public ContentLikeCommentResponse getContent(String memberId, Long contentId) {
+    public CommentsResponse getComments(String memberId, Long contentId) {
         final Member member = findMember(memberId);
         final Content content = findContent(contentId);
         final Baby baby = content.getBaby();
 
         final Relation relation = findRelation(member, baby);
         final RelationGroup relationGroup = relation.getRelationGroup();
+        final List<Comment> comments = findSharedComments(content, baby, relationGroup);
 
-        final List<Like> likes = likeRepository.findAllByContent(content);
-        final List<Comment> comments = commentRepository.findAllByContent(content);
-
-        if (!relationGroup.isFamily()) {
-            final List<Like> sharedLikes = findSharedLikes(baby, relationGroup, likes);
-            final List<Comment> sharedComments = findSharedComments(baby, relationGroup, comments);
-            return getContentLikeCommentResponse(content, sharedLikes, sharedComments);
-        }
-
-        return getContentLikeCommentResponse(content, likes, comments);
-    }
-
-    private List<Like> findSharedLikes(Baby baby, RelationGroup relationGroup, List<Like> likes) {
-        return likes.stream()
-                .filter(like -> {
-                    final Relation likeMemberRelation = findRelation(like.getMember(), baby);
-                    return relationGroup.canShare(likeMemberRelation.getRelationGroup());
-                })
-                .toList();
+        return getCommentsResponse(content, comments);
     }
 
-    private List<Comment> findSharedComments(Baby baby, RelationGroup relationGroup, List<Comment> comments) {
-        return comments.stream()
+    private List<Comment> findSharedComments(Content content, Baby baby, RelationGroup relationGroup) {
+        return commentRepository.findAllByContent(content)
+                .stream()
                 .filter(comment -> {
                     final Relation commentMemberRelation = findRelation(comment.getOwner(), baby);
-                    return relationGroup.canShare(commentMemberRelation.getRelationGroup());
+                    final boolean canShareCommentMemberGroup = relationGroup.canShare(
+                            commentMemberRelation.getRelationGroup());
+                    if (canShareCommentMemberGroup) {
+                        final Optional<Tag> tag = tagRepository.findByComment(comment);
+                        if (tag.isPresent()) {
+                            final Relation tagMemberRelation = findRelation(tag.get().getTagMember(), baby);
+                            return relationGroup.canShare(tagMemberRelation.getRelationGroup());
+                        }
+                    }
+                    return canShareCommentMemberGroup;
```

Comment:
```suggestion
                    final boolean canShareCommentMemberGroup = relationGroup.canShare(
                            commentMemberRelation.getRelationGroup());

                    if (!canShareCommentMemberGroup) {
                        return false;
                    }

                    final Optional<Tag> tag = tagRepository.findByComment(comment);
                    if (tag.isEmpty()) {
                        return true;
                    }

                    final Relation tagMemberRelation = findRelation(tag.get().getTagMember(), baby);
                    return relationGroup.canShare(tagMemberRelation.getRelationGroup());
```

중첩 if문보다는 이런식으로 변경하는게 나을 것 같은데 어떻게 생각하시나요?

Reply:
좋네요 조금 더 명확하게 알아볼 수 있겠네요!